lib/Site.js

```diff
@@ -64,6 +61,10 @@ function Site(rootPath, outputPath) {
   this.pageTemplatePath = path.join(__dirname, TEMPLATE_ROOT_FOLDER_NAME, PAGE_TEMPLATE_NAME);
   this.pageTemplate = ejs.compile(fs.readFileSync(this.pageTemplatePath, 'utf8'));
   this.pageModels = [];
+
+  // Addressable paths and front matter data array
+  this.addressablePaths = [];
+  this.frontMatterData = [];
```

Comment:
Is maintaining this state necessary? Why not assign locally in `writeSiteData`?

Reply:
Yes. I'm accessing it from `Site.prototype.regenerateAffectedPages` and `Site.prototype.generatePages`.

Reply:
Why not assign locally in `writeSiteData`?
It's used immediately and ever only used once each time it is generated.